File: Daily Challanges/2026-06-June/2026-06-10.py
from math import factorial


def getItineraryCount(stops):
    n = len(stops)
    return (2 * n - 3) * factorial(n)


print(getItineraryCount(["library", "park"]))
print(getItineraryCount(["library", "park", "arcade"]))
print(getItineraryCount(["library", "park", "arcade", "store"]))
print(getItineraryCount(["library", "park", "arcade", "store", "cafe"]))
print(
    getItineraryCount(
        ["library", "park", "arcade", "store", "cafe", "market", "museum"]
    )
)

File: Daily Challanges/2026-06-June/2026-06-10.js
function getItineraryCount(stops) {
  const n = stops.length;
  let fact = 1;
  for (let i = 2; i <= n; i++) fact *= i;
  return (2 * n - 3) * fact;
}

console.log(getItineraryCount(["library", "park"]));
console.log(getItineraryCount(["library", "park", "arcade"]));
console.log(getItineraryCount(["library", "park", "arcade", "store"]));
console.log(getItineraryCount(["library", "park", "arcade", "store", "cafe"]));
console.log(
  getItineraryCount([
    "library",
    "park",
    "arcade",
    "store",
    "cafe",
    "market",
    "museum",
  ]),
);
